EIPS/eip-3540.md

```diff
@@ -47,44 +47,40 @@ In order to guarantee that every EOF-formatted contract in the state is valid, w
 
 ### Remarks
 
-The *initcode* is the code executed in the context of the *create* transaction, `CREATE`, or `CREATE2` instructions. The *initcode* returns *code* (via the `RETURN` instruction), which is inserted into the account. See section 7 ("Contract Creation") in the Yellow Paper for more information.
+If code starts with the `MAGIC`, it is considered to be EOF formatted, otherwise it is considered to be *legacy* code. For clarity, the `MAGIC` together with a version number *n* is denoted as the *EOFn prefix*, e.g. *EOF1 prefix*.
 
-The opcode `0xEF` is currently an undefined instruction, therefore: *It pops no stack items and pushes no stack items, and it causes an exceptional abort when executed.* This means *initcode* or already deployed *code* starting with this instruction will continue to abort execution.
+EOF-formatted contracts are created using new instructions `CREATE3`, `CREATE4` and `RETURNCONTRACT`. They are introduced in a separate EIP-???? (TBD, refer to the [Megaspec doc](https://github.com/ipsilon/eof/blob/main/spec/eof.md) until then).
+
+The opcode `0xEF` is currently an undefined instruction, therefore: *It pops no stack items and pushes no stack items, and it causes an exceptional abort when executed.* This means legacy *initcode* or already deployed legacy *code* starting with this instruction will continue to abort execution.
 
 Unless otherwised specified, all integers are encoded in big-endian byte order.
 
 ### Code validation
 
-We introduce *code validation* for new contract creation. To achieve this, we define a format called EVM Object Format (EOF), containing a version indicator, and a ruleset of validity tied to a given version.
-
-At `block.number == HF_BLOCK` new contract creation is modified:
-
-- if *initcode* or *code* starts with the `MAGIC`, it is considered to be EOF formatted and will undergo validation specified in the following sections,
-- else if *code* starts with `0xEF`, creation continues to result in an exceptional abort (the rule introduced in EIP-3541),
-- otherwise code is considered *legacy code* and the following rules do not apply to it.
+We introduce *code validation* for new contract creation, which EOF formatted code will undergo. To achieve this, we define a format called EVM Object Format (EOF), containing a version indicator, and a ruleset of validity tied to a given version.
```

Comment:
I think it's fine to leave like before, in you version "EOF" is used before being defined in next sentence.
```suggestion
We introduce *code validation* for new contract creation. To achieve this, we define a format called EVM Object Format (EOF), containing a version indicator, and a ruleset of validity tied to a given version.
```